shame_script.py

```diff
@@ -39,69 +61,80 @@ async def safe_send(channel: discord.TextChannel, message: str) -> discord.Messa
 
 
 # Function to get all tasks with a specific label and due today or overdue
-def get_tasks(api_token, label_name):
+async def get_tasks(
+    api_token: str, label_name: str, session: aiohttp.ClientSession
+) -> List[Task] | None:
     headers = {"Authorization": f"Bearer {api_token}"}
 
     # Get all tasks
     params = {"filter": f"(today | overdue) & !@{label_name}"}
-    response = requests.get(TODOIST_API, params=params, headers=headers)
-    return response.json()
+    async with session.get(TODOIST_API, params=params, headers=headers) as response:
+        logger.debug("Request from todoist")
+        return await response.json()
 
 
-def add_label(tasks, api_token, label_name):
+async def add_label(
+    task_list: List[Task],
+    api_token: str,
+    label_name: str,
+    session: aiohttp.ClientSession,
+):
     headers = {
         "Authorization": f"Bearer {api_token}",
         "Content-Type": "application/json",
     }
 
     # First, we need to get the ID of the label from the label name
-    labels_response = requests.get(
+    async with session.get(
```

Comment:
what is the impact of async with session here?

Reply:
> Is this the aiohttp version of requests.get() ?

yes

this should perform better than requests  as one http session is created and shared over all the interactions